### PR TITLE
close all maps when agent is terminating

### PR DESCRIPTION
**Pull request by msherif1234**
## Description

we can have stale map when the agent pod has been terminated
## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [] Will this change affect NetObserv / Network Observability operator? If not, you can ignore the rest of this checklist.
* [ ] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [x] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: pkg/ebpf/tracer.go [hunk 1]
```diff
@@ -323,7 +327,15 @@ func (m *FlowFetcher) Close() error {
...
 			errs = append(errs, err)
 		}
-		m.objects = nil
+		if err := m.objects.DnsFlows.Close(); err != nil {
+			errs = append(errs, err)
+		}
+		if err := m.objects.FlowSequences.Close(); err != nil {
+			errs = append(errs, err)
+		}
+		if len(errs) == 0 {
```
Comment (jotak) [inline pkg/ebpf/tracer.go:336]: why do we need to keep the objects reference if there are errors?

Comment (msherif1234) [reply pkg/ebpf/tracer.go:336]: in case I wanted to retry

**Comment by jotak** [comment]
code lgtm

/lgtm
Is that something with a bug / reproducible steps for QE? If not, can you update the description template for no-qe ?

**Comment by msherif1234** [comment]
> code lgtm
> 
> /lgtm Is that something with a bug / reproducible steps for QE? If not, can you update the description template for no-qe ?

No bug for it I saw it while doing some tests and while checking the code notice a couple of maps not freed up when the prog is closed

**Comment by jpinsonneau** [comment]
LGTM too, merging this as it's no qe task

**Pull request merged**